### PR TITLE
Support adding roles for PowerMax

### PR DESCRIPTION
**Pull request by atye**
# Description

This PR adds support for creating a role for a PowerMax system.

This also fixes an issue if you create PowerMax storage and then Powerflex storage, the PowerMax systems would be listed under PowerFlex storage.

To validate a PowerMax role, we verify that the system ID exists as an authorized storage system and verify that the specified Storage Resrouce Pool exists on the PowerMax.

The cobra commands have been refactored in favor of contstructors rather than global variables. This fixes an issue in test runs where we want to update/set various flags on the commands.

# Issues

List the issues impacted by this PR:

| Issue ID |
| -------- |
|     #83      |

# Checklist:

- [x] I have performed a self-review of my own changes.

```
# make test
docker run --rm -it -v /git/karavi-authorization/policies:/policies/ openpolicyagent/opa test -v /policies/
data.karavi.authz.url.test_get_api_login_allowed: PASS (1.792405ms)
data.karavi.authz.url.test_post_proxy_refresh_token_allowed: PASS (711.846µs)
data.karavi.authz.url.test_get_api_version_allowed: PASS (1.538507ms)
data.karavi.authz.url.test_get_system_instances_allowed: PASS (1.241773ms)
data.karavi.authz.url.test_get_storagpool_instances_allowed: PASS (944.188µs)
data.karavi.authz.url.test_post_volume_instances_allowed: PASS (880.791µs)
data.karavi.authz.url.test_get_volume_instance_allowed: PASS (829.917µs)
data.karavi.authz.url.test_post_volume_instances_queryIdByKey_allowed: PASS (1.043505ms)
data.karavi.authz.url.test_get_system_sdc_allowed: PASS (946.403µs)
data.karavi.authz.url.test_post_volume_add_sdc_allowed: PASS (762.907µs)
data.karavi.authz.url.test_post_volume_remove_sdc_allowed: PASS (5.866155ms)
data.karavi.authz.url.test_post_volume_remove_allowed: PASS (628.847µs)
data.karavi.volumes.create.test_small_request_allowed: PASS (2.035073ms)
data.karavi.volumes.create.test_large_request_not_allowed: PASS (375.062µs)
--------------------------------------------------------------------------------
PASS: 14/14
go test -count=1 -cover -race -timeout 30s -short ./...
?       karavi-authorization/cmd/karavictl      [no test files]
ok      karavi-authorization/cmd/karavictl/cmd  5.433s  coverage: 60.9% of statements
ok      karavi-authorization/cmd/proxy-server   0.061s  coverage: 0.7% of statements
?       karavi-authorization/cmd/sidecar-proxy  [no test files]
?       karavi-authorization/cmd/tenant-service [no test files]
ok      karavi-authorization/deploy     0.101s  coverage: 84.6% of statements
?       karavi-authorization/internal/decision  [no test files]
ok      karavi-authorization/internal/powerflex 9.125s  coverage: 88.9% of statements
ok      karavi-authorization/internal/proxy     6.454s  coverage: 68.0% of statements
ok      karavi-authorization/internal/quota     0.514s  coverage: 92.6% of statements
ok      karavi-authorization/internal/roles     0.076s  coverage: 96.0% of statements
ok      karavi-authorization/internal/tenantsvc 3.907s  coverage: 82.9% of statements
ok      karavi-authorization/internal/token     0.045s  coverage: 90.0% of statements
ok      karavi-authorization/internal/web       0.034s  coverage: 24.1% of statements
?       karavi-authorization/pb [no test files]
```

### CODE REVIEW

**Comment by ioeicicdsvc** [comment]
@atye: Thank you for submitting your pull request!

Please make sure you have followed our contributing guidelines. We will review it as soon as possible.

------------------------------------------------------------------------------------------------------------------------
We want your feedback! If you have any questions or suggestions regarding our contributing process/workflow, please reach out to us at karavi@dell.com.

**Review by hoppea2 — DISMISSED** [review]

**Review by sharmilarama — CHANGES_REQUESTED** [review]
LGTM. Please look in to failed tests.

**Review by medegw01 — APPROVED** [review]

**Review by hoppea2 — APPROVED** [review]

**Comment by atye** [comment]
> LGTM. Please look in to failed tests.

Tests have been fixed via refactoring of the cobra commands with constructors rather than global variables.

**Review by ianbibby — APPROVED** [review]
👍

**Review by ianbibby — CHANGES_REQUESTED** [review]
Changes look good.  There are some failing checks that just need to be addressed, e.g. gosec.

**Comment by atye** [comment]
> Changes look good. There are some failing checks that just need to be addressed, e.g. gosec.

Fixed the gosec issue. There is also a security issue in jwt-go which is fixed in https://github.com/dgrijalva/jwt-go/releases/tag/v4.0.0-preview1. I'll update the package.

File: cmd/karavictl/cmd/rolebinding_delete_test.go [hunk 1]
```diff
@@ -70,6 +73,9 @@ func TestRolebindingDelete(t *testing.T) {
 		}
 		var gotOutput bytes.Buffer
 
+		rootCmd := NewRootCmd()
+		//deleteRoleBindingCmd := NewDeleteRoleBindingCmd()
```
Comment (sharmilarama) [inline cmd/karavictl/cmd/rolebinding_delete_test.go:77]: remove comment

Comment (atye) [reply cmd/karavictl/cmd/rolebinding_delete_test.go]: Fixed.

**Review by medegw01 — APPROVED** [review]

**Review by ianbibby — APPROVED** [review]

**Review by sharmilarama — APPROVED** [review]
LGTM

**Pull request merged**